tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -113,7 +113,7 @@ void start(String[] args) throws IOException {
                 record = new ProducerRecord<>(topicName, payload);
 
                 long sendStartMs = System.currentTimeMillis();
-                Callback cb = stats.nextCompletion(sendStartMs, payload.length, stats);
+                cb = stats.nextCompletion(sendStartMs, payload.length);
```

Comment:
`nextCompletion` does not increase `iteration` now, so we can remove the method and then just create callback here.